crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -609,6 +607,21 @@ mod tests {
         let _ = unsafe { BlobVec::new(item_layout, Some(drop), 0) };
     }
 
+    #[test]
+    #[should_panic(expected = "ZST capacity overflow")]
+    fn blob_vec_zst_size_overflow() {
+        // SAFETY: working with `()` which is non-drop ZST.
+        unsafe {
```

Comment:
Preferably each unsafe call should be in it's own unsafe block with their own safety comment directly corresponding to their safety invariants.